### ISSUE TITLE
mean of Vector{Float16} exceeds the maximum value of the vector

### ISSUE THREAD

**Issue by remi-garcia** [JuliaLang/julia#54095]
```julia
julia> versioninfo()
Julia Version 1.6.7
Commit 3b76b25b64 (2022-07-19 15:11 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: 13th Gen Intel(R) Core(TM) i7-1365U
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, goldmont)
```

MWE:
```julia
using Statistics

x = Float16[7.484, 7.477, 7.484, 7.44, 7.5, 7.473, 7.473, 7.5, 7.496, 7.48, 7.5, 7.496, 7.49, 7.492, 7.48, 7.49, 7.49, 7.496, 7.477, 7.49, 7.48, 7.504, 7.5, 7.496, 7.49, 7.49, 7.5, 7.48, 7.496, 7.492, 7.504, 7.508, 7.5, 7.492, 7.484, 7.492, 7.477, 7.496, 7.477, 7.48, 7.484, 7.477, 7.5, 7.484, 7.49, 7.51, 7.5, 7.492, 7.504, 7.49, 7.484, 7.49, 7.49, 7.504, 7.48, 7.504, 7.49, 7.496, 7.492, 7.508, 7.5, 7.492, 7.48, 7.496, 7.492, 7.492, 7.49, 7.46, 7.48, 7.508, 7.48, 7.5, 7.484, 7.457, 7.492, 7.477, 7.49, 7.49, 7.49, 7.457, 7.49, 7.496, 7.473, 7.48, 7.46, 7.496, 7.492, 7.484, 7.48, 7.5, 7.492, 7.492, 7.484, 7.47, 7.492, 7.492, 7.496, 7.453, 7.48, 7.504, 7.49, 7.492, 7.504, 7.48, 7.5, 7.46, 7.508, 7.496, 7.484, 7.48, 7.496, 7.508, 7.496, 7.465, 7.49, 7.508, 7.48, 7.484, 7.465, 7.492, 7.492, 7.484, 7.504, 7.47, 7.473, 7.48, 7.504, 7.516, 7.51, 7.496, 7.492, 7.477, 7.48, 7.473, 7.492, 7.496, 7.492, 7.496, 7.484, 7.49, 7.484, 7.496, 7.508, 7.473, 7.508, 7.473, 7.496, 7.484, 7.47, 7.496, 7.504, 7.477, 7.484, 7.47, 7.492, 7.49, 7.484, 7.496, 7.473, 7.49, 7.473, 7.477, 7.492, 7.48, 7.477, 7.48, 7.48, 7.5, 7.508, 7.484, 7.492, 7.484, 7.492, 7.48, 7.492, 7.484, 7.47, 7.496, 7.492, 7.465, 7.46, 7.47, 7.48, 7.484, 7.504, 7.477, 7.508, 7.473, 7.508, 7.496, 7.504, 7.5, 7.477, 7.492, 7.465, 7.504, 7.504, 7.496, 7.484, 7.492, 7.492, 7.492, 7.492, 7.477, 7.492, 7.5, 7.48, 7.477, 7.477, 7.496, 7.5, 7.508, 7.484, 7.49, 7.496, 7.473, 7.473, 7.496, 7.496, 7.48, 7.473, 7.48, 7.48, 7.473, 7.5, 7.473, 7.473, 7.508, 7.49, 7.484, 7.477, 7.484, 7.496, 7.484, 7.523, 7.504, 7.465, 7.465, 7.48, 7.465, 7.492, 7.496, 7.5, 7.477, 7.484, 7.46, 7.49, 7.5, 7.49, 7.516, 7.492, 7.492, 7.48, 7.465, 7.484, 7.484, 7.492, 7.504, 7.465, 7.48, 7.477, 7.49, 7.496, 7.5, 7.5, 7.492, 7.496, 7.49, 7.496, 7.516, 7.49, 7.492, 7.5, 7.516, 7.47, 7.496, 7.48, 7.5, 7.477, 7.492, 7.51, 7.484, 7.49, 7.496, 7.48, 7.5, 7.496, 7.477, 7.48, 7.496, 7.484, 7.473, 7.49, 7.504, 7.492, 7.492, 7.453, 7.49, 7.496, 7.5, 7.492, 7.49, 7.49, 7.484, 7.496, 7.49, 7.49, 7.47, 7.508, 7.492, 7.492, 7.492, 7.48, 7.48, 7.473, 7.492, 7.477, 7.48, 7.492, 7.492, 7.492, 7.457, 7.48, 7.508, 7.477, 7.47, 7.492, 7.484, 7.48, 7.492, 7.46, 7.477, 7.465, 7.508, 7.473, 7.49, 7.48, 7.484, 7.477, 7.496, 7.47, 7.52, 7.48, 7.508, 7.496, 7.484, 7.49, 7.48, 7.508, 7.477, 7.473, 7.504, 7.492, 7.477, 7.49, 7.492, 7.477, 7.484, 7.504, 7.5, 7.477, 7.49, 7.5, 7.473, 7.5, 7.492, 7.46, 7.492, 7.496, 7.473, 7.484, 7.49, 7.496, 7.477, 7.477, 7.473, 7.53, 7.5, 7.496, 7.477, 7.48, 7.473, 7.496, 7.492, 7.477, 7.457, 7.484, 7.47, 7.496, 7.48, 7.473, 7.49, 7.47, 7.516, 7.477, 7.46, 7.49, 7.484, 7.492, 7.477, 7.477, 7.48, 7.49, 7.49, 7.465, 7.473, 7.484, 7.473, 7.492, 7.477, 7.477, 7.492, 7.49, 7.48, 7.48, 7.492, 7.48, 7.492, 7.49, 7.48, 7.473, 7.492, 7.477, 7.5, 7.496, 7.49, 7.496, 7.47, 7.484, 7.457, 7.48, 7.49, 7.484, 7.473, 7.48, 7.477, 7.49, 7.473, 7.496, 7.473, 7.465, 7.48, 7.473, 7.465, 7.473, 7.492, 7.477, 7.508, 7.48, 7.48, 7.49, 7.484, 7.473, 7.492, 7.49, 7.484, 7.496, 7.49, 7.492, 7.473, 7.492, 7.49, 7.484, 7.51, 7.477, 7.5, 7.47, 7.465, 7.496, 7.504, 7.492, 7.48, 7.477, 7.492, 7.465, 7.484, 7.492, 7.484, 7.49, 7.492, 7.47, 7.49, 7.5, 7.49, 7.492, 7.484, 7.504, 7.504, 7.516, 7.5, 7.49, 7.496, 7.48, 7.496, 7.484, 7.492, 7.484, 7.508, 7.5, 7.496]

println(mean(x) > maximum(x)) #true
```

To get the correct (at least I think so) mean, I do `Float16.(mean(BigFloat.(x)))`

**Comment by KristofferC** [comment]
Seems to be fixed in Julia 1.10.

```
julia> mean(x)
Float16(7.496)

julia> Float16.(mean(BigFloat.(x)))
Float16(7.49)

julia> VERSION
v"1.10.2"
```

**Comment by KristofferC** [comment]
Maybe fixed by https://github.com/JuliaStats/Statistics.jl/pull/150
